content/guides/token-extensions/transfer-hook.md

```diff
@@ -1549,6 +1549,146 @@ Running tests...
   4 passing (5s)
 ```
 
+### Using token account data in transfer hook
+
+Sometimes you may want to use account data to derive additional accounts in the
+extra account metas. This is useful if, for example, you want to use the token
+account's owner as a seed for a PDA.
+
+When creating the ExtraAccountMeta you can use the data of any account as an
+extra seed. In this case we want to derive a counter account from the token
+account owner and the string 'counter'.
+
+This is how you set it up in the InitializeExtraAccountMetaList trait.
+
+```rust
+// Define extra account metas to store on extra_account_meta_list account
+impl<'info> InitializeExtraAccountMetaList<'info> {
+    pub fn extra_account_metas() -> Result<Vec<ExtraAccountMeta>> {
+        Ok(
+            vec![
+                ExtraAccountMeta::new_with_seeds(
+                    &[
+                        Seed::Literal {
+                            bytes: b"counter".to_vec(),
+                        },
+                        Seed::AccountData { account_index: 0, data_index: 32, length: 32 },
+                    ],
+                    false, // is_signer
+                    true // is_writable
+                )?
+            ]
+        )
+    }
+}
+```
+
+Let's look at the token account struct to understand how the account data is
+stored. Below is an example of a token account structure. So we can take 32
+bytes at position 32 to 64 as the owner of the token account, which is at
+'account_index: 0'.
```

Comment:
Could you clarify what you mean by "which is at
'account_index: 0'"?

We are getting the address of the `owner`, which is bytes 32-64 in the Account struct, but not sure what 'account_index: 0' means.